### PR TITLE
[15] fix(manager): only load drop box when we need it/have perms

### PR DESCRIPTION
**Pull request by davidlougheed**

### CODE REVIEW

**Review by v-rocheleau — CHANGES_REQUESTED** [review]
Looking good, minor questions.

File: src/components/manager/ManagerDropBoxContent.js [hunk 1]
```diff
@@ -283,7 +284,13 @@ const DROP_BOX_ROOT_KEY = "/";
 const ManagerDropBoxContent = () => {
     const dispatch = useDispatch();
 
-    const {permissions, hasAttempted} = useResourcePermissions(RESOURCE_EVERYTHING) ?? {};
+    const {
+        permissions,
+        isFetching: isFetchingPermissions,
+        hasAttempted: hasAttemptedPermissions,
+    } = useResourcePermissions(RESOURCE_EVERYTHING) ?? {};
```
[the remark this answers is not in the document]
Comment (davidlougheed) [reply src/components/manager/ManagerDropBoxContent.js]: not needed

File: src/components/manager/hooks.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { RESOURCE_EVERYTHING } from "../../lib/auth/resources";
+import { viewDropBox } from "../../lib/auth/permissions";
+import { fetchDropBoxTreeOrFail } from "../../modules/manager/actions";
+
+export const useFetchDropBoxContentsIfAllowed = () => {
+    const dispatch = useDispatch();
+    const { hasPermission } = useHasResourcePermission(RESOURCE_EVERYTHING, viewDropBox);
+    useEffect(() => {
+        dispatch(fetchDropBoxTreeOrFail());
+    }, [dispatch, hasPermission]);
```
[the remark this answers is not in the document]
Comment (davidlougheed) [reply src/components/manager/hooks.js:17]: missing an if statement which I added + added a comment

**Review by v-rocheleau — APPROVED** [review]
Lgtm!

**Pull request merged**